tests/manage/monitoring/conftest.py

```diff
@@ -0,0 +1,130 @@
+import logging
+import pytest
+import threading
+import time
+
+from ocs_ci.ocs import constants, defaults, ocp
+from ocs_ci.utility.prometheus import PrometheusAPI
+
+
+logger = logging.getLogger(__name__)
+
+
+def measure_operation(
+        operation, minimal_time=None, metadata=None, measure_after=False):
+    """
+    Get dictionary with keys 'start', 'stop', 'metadata' and 'result' that
+    contain information about start and stop time of given function and its
+    result.
+
+    Args:
+        operation (function): Function to be performed.
+        minimal_time (int): Minimal number of seconds to run, it can be more
+            based on given operation.
+        metadata (dict): This can contain dictionary object with information
+            relevant to test (e.g. volume name, operating host, ...).
+        measure_after (bool): Determine if time measurement is done before or
+            after the operation returns its state. This can be useful e.g.
+            for capacity utilization testing where operation fills capacity
+            and utilized data are measured after the utilization is completed.
+
+    Returns:
+        dict: contains information about `start` and `stop` time of given
+            function and its `result` and provided `metadata`.
+    """
+    def prometheus_log(info, alert_list):
+        """
+        Log all alerts from Prometheus API every 10 seconds.
+
+        Args:
+            run (bool): When this var turns into False the thread stops.
+            alert_list (list): List to be populated with alerts
+        """
+        prometheus = PrometheusAPI()
+        while info['run']:
+            alerts_response = prometheus.get(
+                'alerts',
+                payload={
+                    'silenced': False,
+                    'inhibited': False
+                }
+            )
+            assert alerts_response.ok is True, 'Prometheus API request failed'
+            for alert in alerts_response.json()['data']['alerts']:
+                if alert not in alert_list:
+                    logger.info(f"Adding {alert} to alert list")
+                    alert_list.append(alert)
+            time.sleep(10)
+
+    if not measure_after:
+        start_time = time.time()
+
+    # init logging thread
+    # based on https://docs.python.org/3/howto/logging-cookbook.html#logging-from-multiple-threads
+    info = {'run': True}
+    alert_list = []
+
+    logging_thread = threading.Thread(
+        target=prometheus_log,
+        args=(info, alert_list)
+    )
+    logging_thread.start()
+
+    try:
+        result = operation()
+        if measure_after:
+            start_time = time.time()
+        passed_time = time.time() - start_time
+        if minimal_time:
+            additional_time = minimal_time - passed_time
+            if additional_time > 0:
+                time.sleep(additional_time)
+        stop_time = time.time()
+    except KeyboardInterrupt:
```

Comment:
Please add comments in the log for why this could be terminated by Ctrl+c

Reply:
I added more comments to code and logs. This is handler for situation when user interrupts program from keyboard with ctrl+c.

Reply:
@ebenahar it's vice versa: to make sure that things won't fall apart in case one tries to stop the test run via ctrl-c

Reply:
@mbukatov got you

Reply:
We don't need to assume such a thing.
The tests should be considered as run-in Jenkins

Reply:
please fix

Reply:
I don't understand. Do you mean removing user interruption handling so that there will be thread hanging on machine when user interrupts the test during workload phase?

Reply:
done